LeagueOfTowers/Assets/Scripts/ProjectileScript.cs

```diff
@@ -0,0 +1,171 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Photon.Pun;
+public class ProjectileScript : MonoBehaviourPun, IPunInstantiateMagicCallback
+{
+    //fields
+    private string myName;
+    private float mySpeed;
+    private float myDamage;
+    private TowerScript ownerTower;
+    private MonsterScript targetMonster;
+
+
+    //components
+    private PhotonView view;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        this.view = this.GetComponent<PhotonView>();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if(this.targetMonster != null ){
+            //Debug.Log("moving");
+            if(this.targetMonster.GetIsActive())
+            {
+                this.moveToTargetMonster();
+            }
+            else
+            {
+                DestroyThisProjectile();
+            }
+            
+        }
+        else{
+            DestroyThisProjectile();
+        }
+        // else{
+        //     Debug.Log("not found");
+        // }
+    }
+
+    
+
+    //get/set fields
+    public TowerScript GetOwnerTower()
+    {
+        return this.ownerTower;
+    }
+
+    public int GetProjectileViewID()
+    {
+        return this.view.ViewID;
+    }
+
+    //events
+    public void OnPhotonInstantiate(PhotonMessageInfo info)
+    {
+        //Debug.Log("called!");
+        object[] data = this.gameObject.GetPhotonView().InstantiationData;
+        if(data != null && data.Length == 5){
+            //projectileType == projectileName
```

Comment:
commented code

Reply:
I need to keep that comment to remind me that this projectileType field is also a projectile's name.

Reply:
I see